content/posts/GSoC-week-1.md

```diff
@@ -0,0 +1,46 @@
+
+### What is PyBaMM?
+
+
+![PyBaMM Banner](/Pybamm_banner.png)
+
+
+
+PyBaMM (Python Battery Mathematical Modelling) is a tool for fast and flexible simulations of battery models. It provides an open-source framework for multi-institutional, interdisciplinary collaboration. PyBaMM offers improved collaboration and research impact in battery modelling by creating a modular framework through which either existing or new tools can be combined to solve continuum models for batteries. You can learn more about PyBaMM from the official docs [here](https://pybamm.org/).
+
+
+### What am I working on?
+
+
+PyBaMM is very active with its infrastructure-related changes. Maintainers actively look for the latest changes in the scientific Python community and try to keep up with them. My project deals with improvements related to the testing infrastructure at PyBaMM. PyBaMM currently has all their tests written in `Unittest` style tests. Unittest is a framework included in the Python standard library and is used by many Python-based projects, but it is now slowly getting replaced by `Pytest`. `Pytest` is a popular third-party testing framework known for its simplicity, powerful test discovery, and expressive syntax.
+
+The good news is, `Pytest` supports unittest style tests out of the box. So the existing tests (written in `Unittest` style test) will work with `Pytest`. Even before GSoC, I was working on using `Pytest` to run all the tests. The single biggest advantage is that `Pytest` offers parallel execution of tests using `pytest-xdist` plugin. This drastically improves the time taken to run tests. 
+
+We already merged a PR that allows unit-tests to run with `Pytest`. In the first week, I also managed to run integration tests with it. It was not as straightforward as unit-tests. The main problem was with running integration tests in parallel. Most of the integration tests were dumping the JSON data to the same file. Running them parallel resulted in race conditions and the JSON file was getting corrupted. To overcome this issue we decided to use a `uuid` for each filename being generated during tests. This helped us run all integration tests without any skips in parallel with whopping speed gains. You can look at this PR [here](https://github.com/pybamm-team/PyBaMM/pull/4125). 
+
+In future, my work would be mostly rewriting `Unittest` style tests to `Pytest`. Doing so manually would be tedious so we will look into tools that can help automate the process. We also want to implement property-based testing using `Hypothesis` framework. Documenting this whole process can be helpful for other projects planning such migrations in future. You can read more about my GSoC project [here](https://summerofcode.withgoogle.com/programs/2024/projects/gnFfAnqb).
+
+
+Note: I'll be writing biweekly blogs to keep things updated. If you are interested in this project, you can come back every two weeks to check out the progress. 
```

Comment:
```suggestion
Note: I'll be writing fortnightly blogs to keep things updated. If you are interested in this project, you can come back every two weeks to check out the progress. 
```

"biweekly" is a word that I personally avoid using because of its ambiguous meaning – it can mean either "once every two weeks" or "twice a week". I would suggest using "fortnightly", which means the former.